internal/service/wafv2/rule_group.go

```diff
@@ -79,9 +80,22 @@ func ResourceRuleGroup() *schema.Resource {
 					Computed: true,
 				},
 				"name": {
-					Type:     schema.TypeString,
-					Required: true,
-					ForceNew: true,
+					Type:          schema.TypeString,
+					Optional:      true,
+					Computed:      true,
+					ForceNew:      true,
+					ConflictsWith: []string{"name_prefix"},
+					ValidateFunc: validation.All(
+						validation.StringLenBetween(1, 128),
+						validation.StringMatch(regexache.MustCompile(`^[a-zA-Z0-9-_]+$`), "must contain only alphanumeric hyphen and underscore characters"),
+					),
```

Comment:
Duplicating this was easy enough, I suspect one could be a bit smarter here and extract this into a function, and maybe also check that the prefix is short enough to allow some text to be added, and _maybe_ it also needs to take that into account when building the name, and maybe truncate it at that point?

Let me know what you think.

Reply:
The standard pattern is to check that the string length is between 1 and _nameAttrMaxLength_ `- id.UniqueIDSuffixLength` for `name_prefix`.
I have added this.